### ISSUE TITLE
[bug] aws-sdk-cpp/1.9.234 - broken dependencies

### ISSUE THREAD

**Issue by DevAGCO** [conan-io/conan-center-index#24955]
### Describe the bug

OS: Ubuntu 22.04
Conan version 2.6.0

aws-sdk-cpp package has broken dependencies:
`conan install --require aws-sdk-cpp/1.9.234`
outputs:
```
======== Input profiles ========
Profile host:
[settings]
arch=x86_64
build_type=Release
compiler=gcc
compiler.cppstd=gnu17
compiler.libcxx=libstdc++11
compiler.version=11
os=Linux

Profile build:
 [settings]
arch=x86_64
build_type=Release
compiler=gcc
compiler.cppstd=gnu17
compiler.libcxx=libstdc++11
compiler.version=11
os=Linux


======== Computing dependency graph ========
Graph root
    cli
Requirements
    aws-c-common/0.8.2#b8f8529d9e549fff9be397c36de6af2e - Cache
    aws-c-event-stream/0.2.7#96fce11d8aaa217cac51edac17f237bc - Cache
    aws-sdk-cpp/1.9.234#0ad8828e94ec4025c17631a41f62e629 - Cache
ERROR: Version conflict: Conflict between aws-c-common/0.6.11 and aws-c-common/0.8.2 in the graph.
Conflict originates from aws-c-event-stream/0.2.7

Run 'conan graph info ... --format=html > graph.html' and open 'graph.html' to inspect the conflict graphically.
```
conan graph (remove .txt):
[graph.html.txt](https://github.com/user-attachments/files/16625370/graph.html.txt)

### How to reproduce it

`conan install --require aws-sdk-cpp/1.9.234`

**Comment by jcar87** [comment]
Hi @DevAGCO  - we're undergoing an update of the aws recipes to align the versions of their internal dependencies of the AWS SDK cpp, as per https://github.com/conan-io/conan-center-index/issues/24831

Please accept my apologies and we understand the `aws-sdk-cpp `recipe is quite popular and critical to our users needs, and we did not expect to cause a disruption that lasted this long.

In the meantime, you could try the following to get a non-conflicting graph by reverting to the last well-known state:

save a file, e.g. `aws-sdk-cpp.lock`, with the following contents

```
{
    "version": "0.5",
    "requires": [
        "aws-sdk-cpp/1.9.234#0ad8828e94ec4025c17631a41f62e629%1693958564.31",
        "aws-c-event-stream/0.2.7#b47d721d67f9021cced345ab2327eab4%1699584798.186"
    ],
    "build_requires": [],
    "python_requires": []
}
```

And when invoking `conan install`, pass the following arguments (note the `--lockfile` points to the file you just created)

```
conan install --require aws-sdk-cpp/1.9.234 --lockfile-partial --lockfile=aws-sdk-cpp.lock
```

This should get the graph to solve correctly - may experience missing binaries from Conan Center as OpenSSL was recently updated, but it should all build - please report if this is not the case.

**Comment by danimtb** [comment]
@DevAGCO I have opened a PR solving the conflicts of the dependencies here https://github.com/conan-io/conan-center-index/pull/24965.

The recipe has many requirements that depend on each other and are somehow difficult to maintain.

As a guide for the future, I created an issue at https://github.com/conan-io/conan-center-index/issues/24831 with the details of what versions to use for each SDK version and how to proceed when adding future versions of the SDK.

Hope it helps 😄
